### DISCUSSION TITLE
Why is ansible-lint pre-commit-hook showing errors on upstream directories?

### DISCUSSION THREAD

**Discussion by staticdev** [ansible/ansible-lint#1207] [Q&A] [answered]
I just created a role and set a pre-commit-hook on it. When I run, I get errors from one of the requirements, but not to my role. Is this behavior normal or is it a bug?

eg. /home/user/myrole - I execute the hook here, it give an error on/home/user/.ansible/roles/somedependecy

If this is not a bug, is there a way I prevent ansible-lint to looking in upper folders?

**Comment by ssbarnea** [comment, marked as answer]
That is how it works as it uses ansible to load the code. For now the only way to avoid these errors is to add these paths to the ignore paths.

If we would just ignore these files we would risk giving false positives about the code. I really doubt this will change any time soon.
